### PR TITLE
refactor(payment): INT-6964 Sunset Amazon Pay V1

### PR DESCRIPTION
**Pull request by mauricio-sg**
## What? [INT-6964](https://bigcommercecloud.atlassian.net/browse/INT-6964)
Remove all of the code related to Amazon Pay V1.

## Why?
So that we don’t have to continue maintaining.

## Testing / Proof
CircleCI

## Dependency of
👉 https://github.com/bigcommerce/checkout-sdk-js/pull/1700

@bigcommerce/checkout

[INT-6964]: https://bigcommercecloud.atlassian.net/browse/INT-6964?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ggarcia-bc — APPROVED** [review]

**Comment by bc-lessgarmi** [comment]
💚 Tested!
QE details:
The task for AmazonPay has been validated without issue.

The following was tested:
• Regression Amazon Pay

Please see the attachment word below with the evidence:
https://drive.google.com/drive/folders/1dilqgH7eA5eFwMljWf9lS39aeR04CH4W?usp=share_link

Please let me know any comment or doubt

**Review by Tharaae — APPROVED** [review]
LGTM

**Review by animesh1987 — DISMISSED** [review]

File: packages/core/src/app/embeddedCheckout/createEmbeddedCheckoutSupport.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { CheckoutSupport, NoopCheckoutSupport } from '../checkout';
 import EmbeddedCheckoutSupport from './EmbeddedCheckoutSupport';
 import isEmbedded from './isEmbedded';
 
-const UNSUPPORTED_METHODS = ['afterpay', 'applepay', 'amazon', 'chasepay', 'googlepay', 'klarna', 'masterpass'];
+const UNSUPPORTED_METHODS = ['afterpay', 'applepay', 'chasepay', 'googlepay', 'klarna', 'masterpass'];
```
Comment (animesh1987) [inline packages/core/src/app/embeddedCheckout/createEmbeddedCheckoutSupport.ts:8]: Do we need to add `amazonpay` here?

**Review by bc-peng — APPROVED** [review]
Nice work. 👏

File: packages/core/src/app/embeddedCheckout/createEmbeddedCheckoutSupport.ts (hunk 1 above)
Comment (mauricio-sg) [reply packages/core/src/app/embeddedCheckout/createEmbeddedCheckoutSupport.ts:8]: I think so.

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**